### PR TITLE
Articles

### PR DESCRIPTION
**Pull request by blaisefonseca**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,22 +41,28 @@
     "build": "rm -rf ./dist && rollup -c",
     "build:watch": "rm -rf ./dist && rollup -c --watch",
     "cleanup": "rm -rf ./dist && rm -rf ./node_modules",
-    "cleanup:hard": "rm -rf ./dist && rm -rf ./node_modules && rm -rf ./package-lock.json"
+    "cleanup:hard": "rm -rf ./dist && rm -rf ./node_modules && rm -rf ./package-lock.json",
+    "cp":"cp -r ./dist /Users/blaisefonseca/Projects/absch.cbd.int/node_modules/@bower_components/@scbd/common"
```
Comment (sbilodeau) [inline package.json]: Should drop this line
```suggestion
```

File: src/components/articles/cbd-article.vue [hunk 2]
```diff
@@ -0,0 +1,132 @@
+
+<template>
+   <div style="border:none;margin-top:10px">
+        <div v-if="!loading">
+
+            <div v-if="!hideCoverImage && article && article.coverImage && article.coverImage.url">
+                <cbd-article-cover-image cover-image="article.coverImage"></cbd-article-cover-image>
```
Comment (sbilodeau) [inline src/components/articles/cbd-article.vue]: this component `cbd-article-cover-image` cannot be found

File: src/components/articles/cbd-article.vue [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        <slot name="loading-article">
+            <div v-if="loading">Loading content... <i class="fa fa-spinner fa-spin"></i></div>
+        </slot>
+    </div>
+
+</template>
+
+<script>
+
+import 'css!cdn!npm/@scbd/ckeditor5-build-inline-full@35.0.0/build/content-style.css';
```
Comment (sbilodeau) [inline src/components/articles/cbd-article.vue:36]: shall we import using npm from packages?

File: src/components/articles/cbd-article.vue [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    },
+    data() {
+        return {
+            returnUrl       : window.location.href,
+            hasEditRights   : false,
+            loading         : false
+        }
+    },
+    created() {
+       this.ArticlesApi = new ArticlesApi({ token: this.$auth.strategy.token.get() });
```
Comment (sbilodeau) [inline src/components/articles/cbd-article.vue]: maybe token can be automatic from ArticleApi construuctor if not specified? see api-base class suggestion.

File: src/components/cbd-nav/footer.vue [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        <footer class="footer" style="position: unset;">
+
+            <div id="block-followblock" class="follow-us settings-tray-editable block block-cbd-utility block-follow-block"
+                data-drupal-settingstray="editable">
+
+
+                <div class="container">
+                    <div class="row">
+                        <div class="col-12 col-lg-3 d-flex align-items-center">
+                            <div class="h2 text-white text-uppercase m-0">Follow us</div>
```
Comment (sbilodeau) [inline src/components/cbd-nav/footer.vue:12]: Note to ourself: will need to add support for translations

File: src/index.js [hunk 6]
```diff
@@ -1 +1 @@
-export { default as CopCountdown } from './components/countdown/cop-15.vue';
\ No newline at end of file
+export { default as CopCountdown } from './components/countdown/cop-15.vue';
```
Comment (sbilodeau) [inline src/index.js:1]: I think we can drop COP-15 button and code

File: src/services/api/api-base.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
+import axios from 'axios'
+import { isFunction } from 'lodash'
+
+let sitePrefixUrl = 'https://api.cbd.int';
+
+if(/\.cbd\.int$/i   .test(window.location.hostname)) sitePrefixUrl= 'https://api.cbd.int';
+if(/\.cbddev\.xyz$/i.test(window.location.hostname)) sitePrefixUrl= 'https://api.cbddev.xyz';
+if(/\localhost$/i   .test(window.location.hostname)) sitePrefixUrl= '/';
+
+const cache          = new Map()
```
Comment (sbilodeau) [inline src/services/api/api-base.js]: unsused
```suggestion
```

File: src/services/api/api-base.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+const defaultOptions = { prefixUrl: sitePrefixUrl, timeout  : 30 * 1000 }
```
Comment (sbilodeau) [inline src/services/api/api-base.js]: use Vue global auth to to acces token as defualt behaviour 
```suggestion
const defaultOptions = { 
   prefixUrl:  sitePrefixUrl, 
   timeout  : 30 * 1000,
   token: () => Vue.$auth.strategy.token.get()
}
```

**Review by sbilodeau — CHANGES_REQUESTED** [review]

File: src/components/articles/cbd-article.vue (hunk 4 above)
Comment (blaisefonseca) [reply src/components/articles/cbd-article.vue]: ArticleApi construuctor  calling super(options) in constructor should be fine?

File: src/components/articles/cbd-article.vue (hunk 2 above)
Comment (blaisefonseca) [reply src/components/articles/cbd-article.vue]: fixed

**Review by sbilodeau — APPROVED** [review]

**Pull request merged**